htlcswitch/switch.go

```diff
@@ -66,6 +66,12 @@ var (
 	// ErrUnreadableFailureMessage is returned when the failure message
 	// cannot be decrypted.
 	ErrUnreadableFailureMessage = errors.New("unreadable failure message")
+
+	// circularRouteErrTemplate is the string used to return errors to the
+	// router when htlcs fail due to circular routes. It is declared here
+	// to allow for exact error matching in tests.
+	circularRouteErrTemplate = "incoming HTLC(%x) disallowed due to " +
+		"circular forward over: %v "
```

Comment:
Downside of formatting is that it is no longer possible to check for this error via `err == circularRouteErr`

Reply:
I guess this is a more general question of how specific we want the errors we surface to the user to be. If we want to include metadata then I think this is an improvement on unit tests where we just copy the error string that is returned inline. If we're happy with more generic user errors, then the failure detail's error string can just be returned.